provider/image.go

```diff
@@ -89,15 +95,50 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 		return "", nil, err
 	}
 
-	// make the build options
+	// get credentials
+	creds, err := config.Load(config.Dir())
+	if err != nil {
+		return "", props, err
+	}
+	creds.CredentialsStore = credentials.DetectDefaultStore(creds.CredentialsStore)
+	auths, err := creds.GetAllCredentials()
+	if err != nil {
+		return "", nil, err
+	}
+	// read auths to a map of authConfigs for the build options to consume
+	authConfigs := make(map[string]types.AuthConfig, len(auths))
+	for k, auth := range auths {
+		authConfigs[k] = types.AuthConfig(auth)
+	}
 
+	//sess, _ := session.NewSession(ctx, "pulumi-docker", "")
+	//dialSession := func(ctx context.Context, proto string, meta map[string][]string) (net.Conn, error) {
+	//	return docker.DialHijack(ctx, "/session", proto, meta)
+	//}
+	//go sess.Run(ctx, dialSession)
+	//defer sess.Close()
+
+	// make the build options
 	opts := types.ImageBuildOptions{
 		Dockerfile: img.Build.Dockerfile,
 		Tags:       []string{img.Name}, //this should build the image locally, sans registry info
 		Remove:     true,
 		//CacheFrom:  img.Build.CachedImages, // TODO: this needs a login, so needs to be handled differently.
 		BuildArgs: build.Args,
 		Version:   build.BuilderVersion,
+
+		AuthConfigs: authConfigs,
+	}
+
+	// Start a session for BuildKit
+	if build.BuilderVersion == defaultBuilder {
+		sess, _ := session.NewSession(ctx, "pulumi-docker", "")
+		dialSession := func(ctx context.Context, proto string, meta map[string][]string) (net.Conn, error) {
+			return docker.DialHijack(ctx, "/session", proto, meta)
+		}
+		go sess.Run(ctx, dialSession)
```

Comment:
I am curious what happens with this fire-and-forget coroutine? Does `sess.Close()` wait until something useful happens or is there a race we're adding here?